E1_TPU_Samples/image_retraining_tpu.py

```diff
@@ -0,0 +1,127 @@
+import os
+import tensorflow as tf
+import tensorflow_datasets as tfds
+import tensorflow_hub as hub
+from tensorflow.python.tpu import tpu_estimator
+from tensorflow.python.tpu import tpu_optimizer
+from tensorflow.python.tpu import tpu_config
+from functools import partial
+from absl import flags, app
+
+flags.DEFINE_string("tpu", None, "TPU Address")
+flags.DEFINE_integer("iterations", 2, "Number of Itertions")
+flags.DEFINE_integer("batch_size", 10, "Size of eahc Batch")
+flags.DEFINE_boolean("use_tpu", True, " Use TPU")
+flags.DEFINE_string(
+    "model_dir",
+    "model_dir/",
+    "Directory to Save the Models and Checkpoint")
+flags.DEFINE_string(
+    "dataset",
+    "horses_or_humans",
+    "TFDS Dataset Name. IMAGE Dimension should be >= 224, channel=3")
+flags.DEFINE_string("data_dir", None, "Directory to Save Data to")
+
+FLAGS = flags.FLAGS
+NUM_CLASSES = None
+
+
+def input_(mode, batch_size, iterations, **kwargs):
+  global NUM_CLASSES
+  dataset, info = tfds.load(
+      kwargs["dataset"],
+      as_supervised=True,
+      split="train" if mode == tf.estimator.ModeKeys.TRAIN else "test",
+      with_info=True,
+      data_dir=kwargs['data_dir']
+  )
+  NUM_CLASSES = info.features['label'].num_classes
+
+  def resize_and_scale(image, label):
+    image = tf.image.resize(image, size=[224, 224])
+    image = tf.cast(image, tf.float32)
+    image = image / tf.reduce_max(tf.gather(image, 0))
+#        label = tf.one_hot(label, info.features['label'].num_classes)
+    return image, label
+
+  dataset = dataset.map(resize_and_scale).shuffle(
+      1000).repeat(iterations).batch(batch_size, drop_remainder=True)
+  return dataset
+
+
+def model_fn(features, labels, mode, params):
+  global NUM_CLASSES
+  assert NUM_CLASSES is not None
+  model = tf.keras.Sequential([
+      hub.KerasLayer("https://tfhub.dev/google/tf2-preview/inception_v3/feature_vector/4",
+                     output_shape=[2048],
+                     trainable=False
+                     ),
+      tf.keras.layers.Dense(NUM_CLASSES, activation="softmax")
+  ])
+  optimizer = None
+  if mode == tf.estimator.ModeKeys.TRAIN:
+    optimizer = tf.optimizers.Adam(params.get("learing_rate", 1e-3))
```

Comment:
Does this work without your hack in TF? Sorry, I can't remember what we landed on.
Maybe add a TODO mentioning the Issue you filed, and say you will refactor once it is fixed. (And if possible, use compat functionality until the Issue is fixed, so someone could just pull these scripts and run them as-is).

Reply:
This doesn't really work without the Hack :(
I can use compat module for this.

Reply:
I added a command line flag `--use_compat`  to switch from hack mode to compat mode. I think that would help. :)